backend/src/main/java/de/btu/swt/backend/version/git_sync/RemoteSynchronizer.java

```diff
@@ -171,28 +174,28 @@ public void commitPullAndPushChanges(File[] files, String commitMessage, User us
         Path indexPath = gitConfig.getPath(version);
         java.io.File index = indexPath.resolve("index.md").toFile();
         index.createNewFile();
-        FileWriter fileWriter = new FileWriter(index);
-        fileWriter.write("| file | raw | generated |\n");
-        fileWriter.write("| ------ | ------ | ------ |\n");
-        for (File file : version.getFiles()) {
-            fileWriter.write("| ");;
-            fileWriter.write(file.getName().replace("_", ""));
-            fileWriter.write(" | ");
-            fileWriter.write("[RAW-Link](");
-            fileWriter.write(indexPath.relativize(file.getPath().toFile().getCanonicalFile().toPath()).toString());
-            fileWriter.write(")");
-            fileWriter.write(" | ");
-            List<String> exports = new LinkedList<>();
-            for (java.io.File export : file.getExports()) {
-                String uri = indexPath.relativize(export.getCanonicalFile().toPath()).toString();
-                String link = getFileExtension(export.getName()).toUpperCase() + "-Link";
-                exports.add("["+link+"]("+uri+")");
+        try (FileWriter fileWriter = new FileWriter(index)) {
+            fileWriter.write("| file | raw | generated |\n");
+            fileWriter.write("| ------ | ------ | ------ |\n");
+            for (File file : version.getFiles()) {
+                fileWriter.write("| ");
+                fileWriter.write(file.getName().replace("_", ""));
+                fileWriter.write(" | ");
+                fileWriter.write("[RAW-Link](");
+                fileWriter.write(indexPath.relativize(file.getPath().toFile().getCanonicalFile().toPath()).toString());
+                fileWriter.write(")");
+                fileWriter.write(" | ");
+                List<String> exports = new LinkedList<>();
+                for (java.io.File export : file.getExports()) {
+                    String uri = indexPath.relativize(export.getCanonicalFile().toPath()).toString();
+                    String link = getFileExtension(export.getName()).toUpperCase() + "-Link";
+                    exports.add("[" + link + "](" + uri + ")");
+                }
+                fileWriter.write(String.join(" &nbsp; ", exports));
+                fileWriter.write(" |\n");
             }
-            fileWriter.write(String.join(" &nbsp; ", exports));
-            fileWriter.write(" |\n");
+            fileWriter.flush();
         }
-        fileWriter.flush();
-        fileWriter.close();
```

Comment:
@pcprinz as far as I can see, the `fileWriter` does not need to be closed anymore as it is incorporated into the try block, right?